src/manager/PositionManager.ts

```diff
@@ -11,8 +11,9 @@ export class PositionManager {
   }
 
```

Comment:
In PositionManager we also have to support two get scene as private(status maintenance) and view. I think `getPositionAndInitIfAbsent(key: string)` and `getPositionReadonly(key: string)` will work.